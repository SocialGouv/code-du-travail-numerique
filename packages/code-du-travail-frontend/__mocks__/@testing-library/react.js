/* eslint-disable import/export */
import { theme } from "@socialgouv/cdtn-ui";
import { render } from "@testing-library/react";
import PropTypes from "prop-types";
import React from "react";
import { ThemeProvider } from "styled-components";

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme.colors}>
      <>{children}</>
    </ThemeProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
