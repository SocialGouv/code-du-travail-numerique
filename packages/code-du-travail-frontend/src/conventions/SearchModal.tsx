import { Modal } from "@socialgouv/cdtn-ui";
import PropTypes from "prop-types";
import React, { useCallback, useEffect, useState } from "react";

import ConventionSearch from "./Search";

type Props = {
  children?: (openModal: (e: React.MouseEvent) => void) => React.ReactNode;
  renderProp?: () => any;
};

export const ConventionModal = ({ children: renderProp }: Props) => {
  const [isModalVisible, setModalVisibility] = useState(false);

  useEffect(function setupListener() {
    function handleOpenModal() {
      setModalVisibility(true);
    }
    window.addEventListener("tooltip-cc-event", handleOpenModal);

    return function cleanupListener() {
      window.removeEventListener("tooltip-cc-event", handleOpenModal);
    };
  });

  const openModal = useCallback((e) => {
    e.preventDefault();
    setModalVisibility(true);
  }, []);

  const closeModal = useCallback(() => {
    setModalVisibility(false);
  }, []);

  return (
    <>
      {renderProp && renderProp(openModal)}
      <Modal
        isOpen={isModalVisible}
        onDismiss={closeModal}
        title="Rechercher votre convention collective"
      >
        <ConventionSearch />
      </Modal>
    </>
  );
};

ConventionModal.propTypes = {
  children: PropTypes.func,
};
