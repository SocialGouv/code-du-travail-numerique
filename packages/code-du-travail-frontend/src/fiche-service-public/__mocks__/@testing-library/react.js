import React from "react";
import PropTypes from "prop-types";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { theme } from "@socialgouv/cdtn-ui";

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme.colors}>
      <>{children}</>
    </ThemeProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
