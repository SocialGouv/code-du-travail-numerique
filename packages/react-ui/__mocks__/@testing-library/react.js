import { render } from "@testing-library/react";
import PropTypes from "prop-types";
import React from "react";
import { ThemeProvider } from "styled-components";

import { colors } from "../../src/theme";

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={colors}>
      <>{children}</>
    </ThemeProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
