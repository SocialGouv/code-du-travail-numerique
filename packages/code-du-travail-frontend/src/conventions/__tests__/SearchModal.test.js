import { render } from "@testing-library/react";
import React from "react";

import { ConventionModal } from "../SearchModal";

// Trouvez votre convention collective

describe("<ConventionModal />", () => {
  it("should render", () => {
    const { container } = render(
      <ConventionModal>
        {(openModal) => (
          <button data-testid="bt" onClick={openModal}>
            voir les conventions
          </button>
        )}
      </ConventionModal>
    );
    expect(container).toMatchSnapshot();
  });

  it("should render a popup when click on button", () => {
    const { baseElement, getByTestId, getByPlaceholderText } = render(
      <ConventionModal>
        {(openModal) => (
          <button data-testid="bt" onClick={openModal}>
            voir les conventions
          </button>
        )}
      </ConventionModal>
    );
    const button = getByTestId("bt");
    button.click();
    const input = getByPlaceholderText(
      /Nom de la convention collective, de l’entreprise ou son SIRET/i
    );
    expect(input).toBeTruthy();
    expect(baseElement).toMatchSnapshot();
  });

  it("should close the modal", async () => {
    const { getByTitle, getByTestId, queryByPlaceholderText } = render(
      <ConventionModal>
        {(openModal) => (
          <button data-testid="bt" onClick={openModal}>
            chercher votre convention
          </button>
        )}
      </ConventionModal>
    );
    const button = getByTestId("bt");
    button.click();
    const el = getByTitle("fermer la modale");
    expect(el).toBeTruthy();
    el.click();
    expect(
      queryByPlaceholderText(
        /Nom de la convention collective, de l’entreprise ou son SIRET/i
      )
    ).toBeNull();
  });
});
