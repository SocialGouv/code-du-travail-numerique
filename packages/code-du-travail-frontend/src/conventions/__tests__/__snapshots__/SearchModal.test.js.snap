// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConventionModal /> should render 1`] = `
<div>
  <button
    data-testid="bt"
  >
    voir les conventions
  </button>
</div>
`;

exports[`<ConventionModal /> should render a popup when click on button 1`] = `
<body>
  <div
    aria-hidden="true"
  >
    <button
      data-testid="bt"
    >
      voir les conventions
    </button>
  </div>
  <reach-portal>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="1"
    />
    <div
      data-focus-lock-disabled="false"
    >
      <div>
        <div
          class="sc-FRrlG hNWtxs"
          data-reach-dialog-overlay=""
        >
          <div
            aria-label="Rechercher votre convention collective"
            aria-modal="true"
            class="sc-fXazdy lhbBKM"
            data-reach-dialog-content=""
            role="dialog"
            tabindex="-1"
          >
            <label
              class="sc-dTSzeu btJghr"
              for="convention-search"
            >
              Renseignez le nom de votre convention collective, le nom de votre entreprise ou son SIRET.
            </label>
            <span
              class="sc-bQCEYZ iOBpgY sc-jLqRGG jXoiKy"
            >
              <input
                class="sc-fXgAZx esWNAb"
                id="convention-search"
                name="q"
                placeholder="Nom de la convention collective, de l’entreprise ou son SIRET"
                type="search"
                value=""
              />
            </span>
            
            <p
              class="sc-bUQyIj eqjFlE"
            >
              Vous ne connaissez pas ou ne trouvez pas votre convention collective ?
              <span
                class="sc-tsGVs cQiTYY"
              >
                Consultez
                 
                <button
                  class="sc-hiKfDv eWfpdP sc-dSnXvR fivlYt"
                  type="button"
                >
                  notre aide
                </button>
                .
              </span>
            </p>
            <button
              class="sc-hiKfDv dNqnKz sc-dvXYtj kMeFfA"
              title="fermer la modale"
            >
              <div
                class="sc-gXfVKN gNNUJi"
              >
                fermer la modale
              </div>
              <svg
                aria-hidden="true"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <line
                  x1="18"
                  x2="6"
                  y1="6"
                  y2="18"
                />
                <line
                  x1="6"
                  x2="18"
                  y1="6"
                  y2="18"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </reach-portal>
</body>
`;
